### PR TITLE
Remove cover query for ps1770

### PR DESCRIPTION
**Pull request by pablopolyte**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Remove cover query for ps1770
| Type?             | bug fix
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes cover image when value is "0"
| How to test?      | Please indicate how to best verify that this PR is correct.
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.

### CODE REVIEW

**Comment by matks** [comment]
Hi @pablopolyte thanks for the PR! Can you give us some context? Why do you suggest to remove the cover query for PS 1770? Is there a change in PS 1.7.7 that requires this?

**Comment by kpodemski** [comment]
It's probably because we're able to get proper cover from .tpl, but I don't see changes related to this

**Comment by sarahdib** [comment]
@matks & @kpodemski it's because I found a bug on the new release of blockwishlist 
https://github.com/PrestaShop/blockwishlist/pull/77#issuecomment-721817405
This PR fix the first bug :)

**Comment by Quetzacoalt91** [comment]
Hello !

We discovered that with one of the wishlist SQL request, we could get a `0` when no image is chosen for a given combination.

The core is able to provide it by itself only if we DO NOT provide this information. As soon as we give a value, even wrong, it does nothing for us: 
https://github.com/PrestaShop/PrestaShop/blob/6a880d31539ab2dcad5be32d7cdc332574a4fe43/classes/Product.php#L5494-L5499

Because of this value zero, it appears that we enter this `if()` block and make the core failing: https://github.com/PrestaShop/PrestaShop/blob/6a880d31539ab2dcad5be32d7cdc332574a4fe43/src/Adapter/Presenter/Product/ProductLazyArray.php#L606-L622

By leaving the core handling it completely, we get rid of this error.

**Comment by kpodemski** [comment]
@Quetzacoalt91 sure, I understand, make sure to check this PR tho:
https://github.com/PrestaShop/blockwishlist/pull/89

As without it y'all make sure to edit .tpl inside a module to get cover based on combination added to wishlist, not default one.

File: src/Search/WishListProductSearchProvider.php [hunk 1]
```diff
@@ -130,9 +130,7 @@ private function getProductsOrCount(
                 )
             ) > 0 AS new'
             );
-            $querySearch->select('
-                (SELECT `id_image` FROM `' . _DB_PREFIX_ . 'product_attribute_image` WHERE `id_product_attribute` = product_attribute_shop.`id_product_attribute`) cover_image_id'
-            );
+
```
Comment (Quetzacoalt91) [inline src/Search/WishListProductSearchProvider.php:133]:
```suggestion
$querySearch->select('IF(pai.`id_image`, pai.`id_image`, NULL) AS cover_image_id');
```

Comment (Quetzacoalt91) [reply src/Search/WishListProductSearchProvider.php:133]: Then later in this file when we join tables:

```patch
+$querySearch->leftJoin('product_attribute_image', 'pai', 'pai.`id_product_attribute` = product_attribute_shop.`id_product_attribute`');
```

**Comment by Quetzacoalt91** [comment]
I see. Because we can't change the behavior on already released versions of PrestaShop, I guess one workaround could be found on the module side.
I'm not sure if we were supposed to find zeros in the column `product_attribute_image.id_image`, but we have to ask MySQL to return null when this happens.

Would the suggestion above this comment help? :arrow_up:

**Comment by kpodemski** [comment]
@Quetzacoalt91 

you can safely remove this query :) to avoid issues with cover:

- remove this additional query from module
- on template level make sure to use `$product.default_image` https://github.com/PrestaShop/PrestaShop/pull/21782/files 

This should be enough.

**Comment by Quetzacoalt91** [comment]
Oh okay ! I didn't get the changes were needed on the template.

Ping @pablopolyte

**Review by NeOMakinG — APPROVED** [review]

**Pull request merged**